locations-list-map.js

```diff
@@ -332,13 +315,73 @@ function onFilterChange(elem) {
-                                    address, name, instructions, accepting, open_accepted);
+                     var marker = addMarkerToMap(map, Number(latitude), Number(longitude),
+                         address, name, instructions, accepting, open_accepted);
+                     markers.push(marker);
                  }
              }
          }
      }
- }
+
+     // Center the map on the nearest markers to the user if possible
+     centerMapToNearestMarkers(map, markers);
+}
+
+function centerMapToNearestMarkers(map, markers) {
+    // First check to see if the user will accept getting their location, if not, silently return
+    if (navigator.geolocation) {
+        navigator.geolocation.getCurrentPosition(function (position) {
+            var user_latlng = new google.maps.LatLng(position.coords.latitude, position.coords.longitude);
+
+            //Compute the distances of all markers from the user
+            var markerDistances = new Map(); // an associative array containing the marker referenced by the computed distance
+            var distances = []; // all the distances, so we can sort and then call markerDistances
+            for (const marker of markers) {
+                var distance = google.maps.geometry.spherical.computeDistanceBetween(marker.position, user_latlng);
+
+                // HACK: In the unlikely event that the exact same distance is computed, add one meter to the distance to give it a unique distance
+                // This could occur if a marker was added twice to the same location.
+                if (markerDistances.has(distance)) { distance = distance + 1; }
+
+                markerDistances[distance] = marker;
+
+                distances.push(distance);
+            }
+
+            // sort the distances and set bounds to closest three
+            distances.sort((a, b) => a - b)
+
+
+            // center the map on the user
+            bounds = new google.maps.LatLngBounds();
+            bounds.extend(user_latlng);
+
+            // Extend the bounds to contain the three closest markers
+            i = 0;
+            while (i < 3) {
+                // Get one of the closest markers
+                var distance = distances[i]
+                var marker = markerDistances[distance];
+
+                // Add to the iterator first just in case something fails later to avoid infinite loop
+                i++;
+
+                marker_lat = marker.position.lat();
+                marker_lng = marker.position.lng();
+
+                loc = new google.maps.LatLng(marker_lat, marker_lng);
+                bounds.extend(loc);
+            }
+            map.fitBounds(bounds);       // auto-zoom
+            map.panToBounds(bounds);     // auto-center
+        }, function () {
+            alert('Could not get user location');
+        });
+    } else {
+        // Browser doesn't support Geolocation
+        alert('Could not get user location');
```

Comment:
Please remove these two alerts before merging.

Reply:
Removed alerts per comment in latest pull request.